src/BenchmarkDotNet/Validators/SdkValidator.cs

```diff
@@ -0,0 +1,142 @@
+﻿using BenchmarkDotNet.Jobs;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace BenchmarkDotNet.Validators
+{
+    public class SdkValidator : IValidator
+    {
+        private readonly ISdkProvider sdkProvider;
+
+        public static readonly SdkValidator Instance = new SdkValidator(new DotNetSdkProvider());
+
+        public SdkValidator(ISdkProvider sdkProvider)
+        {
+            this.sdkProvider = sdkProvider;
+        }
+
+        public bool TreatsWarningsAsErrors => true;
+
+        public IEnumerable<ValidationError> Validate(ValidationParameters validationParameters)
+        {
+            foreach (var benchmark in validationParameters.Benchmarks)
+            {
+                if (benchmark == null || benchmark.Job == null || benchmark.Job.Environment?.Runtime?.RuntimeMoniker == null)
+                {
+                    continue;
+                }
+
+                var runtimeMoniker = benchmark.Job.Environment.Runtime.RuntimeMoniker;
+                if (!IsSdkInstalled(runtimeMoniker))
+                {
+                    yield return new ValidationError(
+                        TreatsWarningsAsErrors,
+                        $"The required SDK for {runtimeMoniker} is not installed",
+                        benchmark);
+                }
+            }
+        }
+
+        private bool IsSdkInstalled(RuntimeMoniker runtimeMoniker)
```

Comment:
This should probably be updated to check the `CustomDotNetCliPath`.

Reply:
Great catch. The path info is passed into `DotNetSdkProvider` but I forgot to update my method there. I will revise it.